### ISSUE TITLE
Interact widget state not stored correctly

### ISSUE THREAD

**Issue by davidbrochart** [jupyter/nbconvert#1214]
On notebook execution, it seems that the widget state is not embedded recursively. Only the top-level widget state is embedded. If the widget is e.g. a `VBox` consisting of several children, none of the children's state is embedded.

The issue can be reproduced by running the following command (`nbconvert 6.0.0a1.dev0`):
```
jupyter nbconvert --execute ipywidgets.ipynb --ExecutePreprocessor.store_widget_state=True --to notebook
```

On the following `ipywidgets.ipynb` notebook:
```python
from ipywidgets import interact
def f(x):
    return x
interact(f, x=10)
```

**Comment by davidbrochart** [comment]
I investigated a bit more, this works fine:
```python
from ipywidgets import IntSlider, Label, VBox
VBox([IntSlider(), Label('hello')])
```
So it looks like the issue is related to `interact` in particular, not to a hierarchy of widgets.

**Comment by maartenbreddels** [comment]
I think it has to do with the output widget. In voila we have a [special executor with a output widget](https://github.com/voila-dashboards/voila/blob/33ab5f301a2b0bf118de905ca432ed1f725d77f6/voila/execute.py#L50) 'frontend', we could consider moving this to nbconvert, what do you think?
I'll investigate later to confirm this.

**Comment by MSeal** [comment]
Posting on both threads -- doing this planned work should fix the Output widget object states, if that is the root cause here: https://github.com/jupyter/nbclient/issues/24

**Comment by MSeal** [comment]
This is now solved in nbclient, which means nbconvert 6.0 alpha has the fix.
